voting/test/it/voting.it.spec.js

```diff
@@ -10,90 +10,384 @@ describe("voting contracts on orbs and ethereum", async () => {
+
+            d1.delegateExplicitly(d4),
+            d7.delegateExplicitly(g3),
+
+            // TODO verify delegation in contract state
+
+            g1.voteOut(v1, v3),
+            g3.voteOut(v3, v4, v5),
+            g2.voteOut(v3),
+            g1.voteOut(v2), // second vote
+            g4.voteOut(),
+            d2.voteOut(v5, v2, v3), // not an guardian
+        ]);
+
+        console.log("Start First Block Election");
+
+        const nextElectionBlockNumber = await electionContracts.setElectionBlockNumber();
+
+        await electionContracts.waitForOrbsFinality(nextElectionBlockNumber);
+
+        await electionContracts.goodSamaritanMirrorsAll(nextElectionBlockNumber);
+        //TODO return something from mirror and assert
+
+        console.log("Done Mirroring");
+
+        const mirrorPeriodEndBlock = nextElectionBlockNumber + options.votingMirrorPeriod + 1;
+
+        await electionContracts.waitForOrbsFinality(mirrorPeriodEndBlock);
+
+        await electionContracts.goodSamaritanProcessesAll();
+        //TODO return something from process and assert
+
+        console.log("Done Processing");
+
+        const winners = await electionContracts.getElectionWinners();
+        expect(winners).to.have.members([v1, v2, v4, v5].map(v => v.orbsAccount.address)); // TODO - which should be voted in???
+
+        // check reward  added to v1 and v2 with time based values.
+        expect(await electionContracts.getOrbsValidatorReward(electionContracts.addressWithoutChecksum(v1.address))).to.be.equal(8533); // +8533
+        expect(await electionContracts.getOrbsValidatorReward(electionContracts.addressWithoutChecksum(v2.address))).to.be.equal(8536); // +8536
+
+        console.log("Done First Election\n");
+
+        console.log("Start Second Block Election");
+        // change election
+        await g1.voteOut(v1);
+        await g2.voteOut(v1);
+        await g3.voteOut(v1);
+        await g4.voteOut(v1);
+
+        await electionContracts.waitForOrbsFinality();
+
+        const nextElectionBlockNumber2 = await electionContracts.setElectionBlockNumber();
+
+        await electionContracts.waitForOrbsFinality(nextElectionBlockNumber2);
+
+        await electionContracts.goodSamaritanMirrorsAll(nextElectionBlockNumber2);
+        //TODO return something from mirror and assert
+
+        console.log("Done Mirroring");
+
+        const mirrorPeriodEndBlock2 = nextElectionBlockNumber2 + options.votingMirrorPeriod + 1;
+
+        await electionContracts.waitForOrbsFinality(mirrorPeriodEndBlock2);
+
+        await electionContracts.goodSamaritanProcessesAll();
+        //TODO return something from process and assert
+
+        console.log("Done Processing");
+
+        const winners2 = await electionContracts.getElectionWinners();
+        expect(winners2).to.have.members([v2, v3, v4, v5].map(v => v.orbsAccount.address)); // TODO - which should be voted in???
+
+        // check reward  added to v1 and v2 with time based values.
```

Comment:
time-based values? but this is a block-based test

Reply:
these are the block based values.

Reply:
the comment says "time based"